### PR TITLE
Procedural Dungeon Generator

### PR DESCRIPTION
**Pull request by dfuchs2**

### CODE REVIEW

**Comment by dfuchs2** [comment]
I'm looking for some advice for what to add. I feel like it isn't complete yet, but I can't decide what I want to add to it.

**Comment by Turtlelord26** [comment]
Do you still want reviews? @dfuchs2

File: Procedural Dungeon Generator [hunk 1]
```diff
@@ -1,29 +1,43 @@
...
-being on a water world with most or all of the environment underwater, in space 
-with no air or gravity, in a snowstorm with very limited visibility but also the 
-ability to track people based on their footprints, on a volcanic planet where 
-staying outside too long without a mask can injure or blind the players, etc. 
-If no specific rules exist for the map you want to make, it will be up to you to 
-create them. In general, map-specific rules should not be a major hinderance to 
-the players, but should make the experience more interesting and challenging. If 
-traversing the map is harder than fighting the enemies, then it won't be a fun game
-for anybody.
+well as any area-specific rules that may apply.
```
Comment (bleehu) [inline Procedural Dungeon Generator:11]: Starting with a planet type is a specific implementation of the more abstract strategy of starting with a theme or idea. A lot of good maps are made by starting with an idea, such as "what if we did a map that was on a living, moving tree? Wouldn't that be cool?" 
Instead, I recommend that your first step in creating a dungeon be setting a goal. How will you know when your dungeon is finished? Dungeoneering is an inherently iterative and recursive process. If you don't set a termination point at the start, then you'll be working on your dungeon forever, and nobody will ever get to see your hard work. Suggested goals are things like "I want this dungeon to be fun for every type of player in Bartle's Taxonomy." Another good list of goals are in the dungeon festival rubric.

File: Procedural Dungeon Generator [hunk 2]
```diff
@@ -38,10 +52,45 @@ not so close as to be unpassable.
...
+own for the encounter. If the enemies in your game have been creating a light-activated, face-
+melting, explosive poison gas, then it's your job to determine exactly how it works.
     
 ==============================
 Add NPCs and Hostiles
 ==============================
     
-        
+        Lastly, your dungeon should have non-playable characters in it. For the sake of interesting 
+story, it's usually a good idea for your characters to have some kind of backstory, whether it's
```
Comment (bleehu) [inline Procedural Dungeon Generator:86]: Story sense is a must-have, not an optional thing for a decent dungeon. If you can't roll investigate at some point in your dungeon to find out something interesting, then you've failed your explorer player types and you've undervalued that particular skill. You've made a chess board, not a dungeon.

File: Procedural Dungeon Generator [hunk 3]
```diff
@@ -1,29 +1,43 @@
...
+well as any area-specific rules that may apply. 
+
+    Some examples include:
+    * The map being on a water world with most or all of the environment underwater. The 
+characters would either have to use underwater breathing gear, or the map-builder would 
+have to write their own breath-holding rules.
+    * In space with no air or gravity. There are rules in place dealing with movement in 
+microgravity, as well as EVA suits that the players could use. 
+    * In a snowstorm with very limited visibility but also the ability to track people based on 
+their footprints. The hunter class might be very useful in a situation such as this, because
```
Comment (bleehu) [inline Procedural Dungeon Generator:20]: What you're driving at here is that a good dungeon has something for every class to excel at. We should say that at some point.

File: Procedural Dungeon Generator [hunk 4]
```diff
@@ -1,29 +1,43 @@
...
+have to write their own breath-holding rules.
+    * In space with no air or gravity. There are rules in place dealing with movement in 
+microgravity, as well as EVA suits that the players could use. 
+    * In a snowstorm with very limited visibility but also the ability to track people based on 
+their footprints. The hunter class might be very useful in a situation such as this, because 
+their awareness might be the reason that the players get to surprise an enemy patrol, instead 
+of the other way around.
+    * On a volcanic planet where staying outside too long without a mask can injure or blind 
+the players. Robots and humanoids with environment helmets would probably be fine here, while 
+anybody else might be forced to stay inside or be hindered by the air quality outside.
```
Comment (bleehu) [inline Procedural Dungeon Generator:25]: Volcanic wasteland would be a super cool battlefield! Great idea!

File: Procedural Dungeon Generator [hunk 5]
```diff
@@ -38,10 +52,45 @@ not so close as to be unpassable.
...
       
-        
+        Now the map needs smaller details to help flesh it out. This is when you add in all
+of the objects that will be scattered about the map. Place tables and chairs in rooms that 
+require them. Create specific items that fit the area they're in, such as dangerous chemicals 
+in a science lab or bunks in a military camp. Computer or electrical panels are also very 
+important. They allow the players to interface with the map in a unique way, and in some 
+cases can be pivotal to the story you are trying to play. When building a map, it's usually 
+a good idea to keep it fairly realistic, but the most important thing to remember is that the 
+players must have fun and that the game be interesting.
```
Comment (bleehu) [inline Procedural Dungeon Generator:62]: You are absolutely right about this! Fun and/or interesting is the most important thing! :+1:

File: Procedural Dungeon Generator [hunk 6]
```diff
@@ -38,10 +52,45 @@ not so close as to be unpassable.
...
-        
+        Lastly, your dungeon should have non-playable characters in it. For the sake of interesting 
+story, it's usually a good idea for your characters to have some kind of backstory, whether it's 
+as simple as the organization they work for, or as complicated as the entire detailed story of their 
+life. Combat personnel should be given weapons and armor based off of their levels, and depending on
+the character, you can also give them other skills and feats to go along with their class. Exactly what
+kind of gear they are given is very much up to you and how you want to run your dungeon, but as a 
+rule of thumb, it's a good idea to make combat fast-paced, tactical, and enjoyable for everybody. 
+
+
```
Comment (bleehu) [inline Procedural Dungeon Generator:93]: ... Where do you talk about loot? You hint at puzzles a couple of times, but don't really explicitly call them out. https://www.youtube.com/watch?v=XWgc20zbRXk&list=PLDPL7seU9mMcBZ8yonxZvyEcFHg8dy1wl

File: Procedural Dungeon Generator [hunk 7]
```diff
@@ -38,10 +52,45 @@ not so close as to be unpassable.
...
+others. It would also be a good idea to think of how the system is secured, as players 
+are able to infiltrate the network.
+    *  A heavily reinforced door might be accessible if the players find the correct wiring in 
+the walls. In general, it's a good idea to have multiple ways for the players to solve a 
+problem. If the there is only one route through your game, it might start to get boring.
+    *  A scrap of paper tacked onto a message board might have the keycode to access a safe 
+where the players might be able to find some loot.
+    *  If you create something that doesn't have rules written for it, feel free to write your 
+own for the encounter. If the enemies in your game have been creating a light-activated, face-
+melting, explosive poison gas, then it's your job to determine exactly how it works.
```
Comment (bleehu) [inline Procedural Dungeon Generator:79]: "light-activated, face-melting, explosive poison gas" :joy: This is just jexcellent! Well said!

File: Procedural Dungeon Generator [hunk 8]
```diff
@@ -38,10 +52,45 @@ not so close as to be unpassable.
...
 ==============================
     
-        
+        Lastly, your dungeon should have non-playable characters in it. For the sake of interesting 
+story, it's usually a good idea for your characters to have some kind of backstory, whether it's 
+as simple as the organization they work for, or as complicated as the entire detailed story of their 
+life. Combat personnel should be given weapons and armor based off of their levels, and depending on
+the character, you can also give them other skills and feats to go along with their class. Exactly what
+kind of gear they are given is very much up to you and how you want to run your dungeon, but as a 
+rule of thumb, it's a good idea to make combat fast-paced, tactical, and enjoyable for everybody.
```
Comment (bleehu) [inline Procedural Dungeon Generator:91]: We should maybe call out specifically that enemies might not use the feats, items or abilities available to the players. It's a common mistake that first-time GMs or Dungeoneers often make. Also, while turns in combat should be quick to keep player's interest, it's ok to vary the pacing during the course of the dungeon to vary the interest curve. https://github.com/bleehu/CX_Design_Guide/blob/master/DevProcesses/welcome.md#interest-curves

**Review by bleehu — CHANGES_REQUESTED** [review]
Can we re-post this article in the Dungeon Master's guide instead of in the middle of the player's handbook? https://github.com/bleehu/CX_Galaxy_Masters_Guide/tree/master/3_Making_Dungeons

**Comment by Turtlelord26** [comment]
@bleehu @dfuchs2 Is this branch safe to delete?

**Pull request closed without merging**